docs/code_guidelines/all.automated_review_guidelines.reference.md

```diff
@@ -0,0 +1,484 @@
+  * [Filing a PR](#filing-a-pr)
+  * [Before requesting review](#before-requesting-review)
+  * [During the review cycle](#during-the-review-cycle)
+- [Code style](#code-style)
+  * [Python code](#python-code)
+    + [Naming](#naming)
+    + [Docstrings](#docstrings)
+    + [Comments](#comments)
+    + [Code design](#code-design)
+    + [Imports](#imports)
+    + [Type annotations](#type-annotations)
+    + [Functions](#functions)
+    + [Scripts](#scripts)
+    + [Logging](#logging)
+    + [Unit tests](#unit-tests)
+    + [Misc](#misc)
+  * [Notebooks](#notebooks)
+    + [General](#general)
+    + [Jupytext](#jupytext)
+    + [Plotting](#plotting)
+  * [Markdowns](#markdowns)
+  * [File system structure](#file-system-structure)
+  * [Spelling](#spelling)
+
+<!-- tocstop -->
+
+This document outlines the rules that all PRs and checked-in code must follow.
+It can serve as a guideline for automated PR reviews.
+
+## PR workflows
+
+### Working in a branch
+
+- The branch should be named following the format
+  `RelatedIssueTag_Normalized_issue_title`
+  - E.g., `HelpersTask123_Provide_branch_name_example`, if the branch is for
+    working on the issue #123 in the `helpers` repo with the title "Provide
+    branch name example"
+- Commit messages should be short and informative
+  - Ideally, they should follow the format
+    `RelatedIssueTag: High-level commit description`
+  - E.g., `HelpersTask123: Add example`
+  - Commit messages should not mention the name of the file that has been
+    changed by the commit
+
+### Filing a PR
+
+- The title of the PR should match the name of the branch
+- The starting post of the PR should briefly describe the content of the PR on a
+  high level
+- The issue related to the PR must be mentioned in the starting post of the PR
+- The PR must not be linked to any issues under the `Development` section
+- At least one reviewer must be assigned under `Reviewers`
+- The PR author must be listed under `Assignees`
+
+### Before requesting review
+
+- All the checks performed by GitHub Actions must pass
+- The branch must be up to date with the master branch
+- There should be no conflicts with the master branch
+- There should be no files checked in by mistake (such as tmp and log files)
+- All checked in files should be checked and formatted by Linter
+- No files larger than 500 KB should be checked in
+- Screenshots should not be used in PR posts to describe the situation or report
+  an error — copy-and-paste instead
+
+### During the review cycle
+
+- Label `PR_for_reviewers` should be present when a review is requested
+- Fixes addressing a review comment should be applied everywhere, not just where
+  the reviewer pointed out the issue
+- After addressing a review comment, the corresponding conversation should be
+  marked as `resolved`
+- After all review comments are resolved, `re-request review` button should be
+  used to request another round of review
+
+## Code style
```

Comment:
This is all very good. Let's keep it here and start experimenting with llm_transform targets.

We can have an executable `ai_reviewer.py` which loads this file and looks for violation (and then can try to fix, but this is difficult due to the large number of false positivies)